eclair-core/src/main/scala/fr/acinq/eclair/crypto/keymanager/LocalChannelKeyManager.scala

```diff
@@ -38,8 +38,23 @@ object LocalChannelKeyManager {
 }
 
 /**
- * This class manages channel secrets and private keys.
- * It exports points and public keys, and provides signing methods
+ * An implementation of [[ChannelKeyManager]] that supports deterministic derivation based on the initial funding pubkey.
+ *
+ * Specifically, there are two paths both of length 8 (256 bits):
+ *   - fundingKeyPath: chosen at random
+ *   - channelKeyPath: sha(fundingPubkey(0))
+ *
+ * The resulting paths looks like so on mainnet:
+ * {{{
+ *  funding:
+ *   - initial/static
+ *     47' / 1' / <fundingKeyPath> / <1 or 0> / 0
+ *   - dynamic
+ *     47' / 1' / <fundingKeyPath> / <1 or 0> / 6 / <index>
+ *
+ *  all others (payment, revocation, htlc, etc.):
+ *     47' / 1' / <channelKeyPath> / <1-5> / <index>
```

Comment:
Please double check this as it is easy to miss something.

Reply:
> That also means we don't have to use 6 for the splice funding txs, if <1-5> are used with a different path anyway

If we use `this/path` for the initial funding tx, and `this/path/<index>` for the subsequent funding txs, it is my understanding that leaking the private key for the initial funding key will compromise all the child keys. So I think we do in fact need a different path.

> it's hard to figure out which paths are already used

Indeed, that's why I inlined the values in my `lightning-kmp` rework: https://github.com/ACINQ/lightning-kmp/blob/62eb5ea0a5a26fd06c69fa499453288a0e2e89dc/src/commonMain/kotlin/fr/acinq/lightning/crypto/LocalKeyManager.kt#L85-L91

> if we want to make it safer we probably need to type this more

Agreed, also the fact that the `KeyManager` has been designed for external signing is a major source of complexity.

Reply:
Simplified in a38095a5f. We are using _sister_ keys, not _child_ keys, so my concern above doesn't apply.